### DISCUSSION TITLE
Taxel location on fingertip

### DISCUSSION THREAD

**Discussion by Yujia-Zhang0913** [robotology/community#668] [Q&A] [answered]
Hi, 
I see that in [https://github.com/robotology/icub-main/blob/master/app/skinGui/conf/positions/left_hand_V2_1.txt](url), the position of taxel on fingertip is zero. Is there any other document that shows the location of taxel? And how do I know which part of fingers have a contact with objects? Or I should consider the whole fingertip as one tactile sensor? Thanks!

**Comment by pattacini** [comment, marked as answer]
Hi @Yujia-Zhang0913 

>Is there any other document that shows the location of taxel?

We do have a description of the 3D location for the version mk1.1. See
- https://github.com/icub-tech-iit/electronics-boards-public/blob/master/other-boards/ft2/docs/3Dtaxelpositions.md

You should have version mk1.0 instead. Although the mapping is not the same, that doc could be helpful anyway.

>And how do I know which part of fingers have a contact with objects? 

You can peruse this [documentation][read-data], which shows how to read the single taxel data.

cc @simeonedussoni 

[read-data]: https://icub-tech-iit.github.io/documentation/tactile_sensors/software/reading_data/?h=taxel#reading-compensated-tactile-data

**Reply by pattacini** [reply]
To expand more on the values found in [**`icub-main/app/skinGui/conf/positions`**](https://github.com/robotology/icub-main/tree/master/app/skinGui/conf/positions):
- The columns report on the x-y-z coordinates of the patch wrt a reference frame, plus the direction of the normal.
- There are many zeros because many patches are not connected.
- The only info reported is related to the patches of the palm, whose reference frame is that of the hand (according to the kinematics).

**Comment by Yujia-Zhang0913** [comment]
Hi @pattacini 
I found that when I read from /icub/skin/right_hand, the numbers range from 0-255.
And I think when I read from /icub/skin/right_hand_comp, the number should range from 0-100 or something else. However, it is always 0 or 100. No any number between 0 and 100. Is this normal or I misunderstand something?

**Reply by pattacini** [reply]
Did you make sure that the binarization option is off in iCubSkinGui?

**Reply by Yujia-Zhang0913** [reply]
Oh, you are right! Thanks!

**Comment by Yujia-Zhang0913** [comment]
Hi @pattacini 
So is there any document about the PCB below?  I want to get the normal directions of taxels on fingertips. Really thanks.
![image](https://github.com/robotology/community/assets/73634487/efef9a33-73e3-4015-bfda-d8dbcbe6bf9b)

**Reply by pattacini** [reply]
Hi @Yujia-Zhang0913 

As said [above](https://github.com/orgs/robotology/discussions/668#discussioncomment-7524289), we do not have similar 3D information for the mk1.0 fingertip, unfortunately.

I'll put this point on the agenda of our next internal meeting to see if we can plan for it.
But considering all our ongoing activities, I'm afraid that it won't be available any time soon.

**Reply by Yujia-Zhang0913** [reply]
I understand, still thanks for helping with this.
